### PR TITLE
Hide asset versioning if it is empty

### PR DESCRIPTION
**Pull request by geerteltink**

### CODE REVIEW

File: src/TwigExtension.php [hunk 1]
```diff
@@ -93,11 +93,13 @@ public function renderUri($name, $parameters = [], $relative = false)
      */
     public function renderAssetUrl($path, $packageName = null, $absolute = false, $version = null)
     {
+        $assetVersion = ($version) ? $version : $this->assetsVersion;
```
Comment (Maks3w) [inline src/TwigExtension.php]: You can write this as `$assetVersion = $version | $this->assetVersion`

**Comment by Maks3w** [comment]
Missing test

File: src/TwigExtension.php [hunk 2]
```diff
@@ -93,11 +93,13 @@ public function renderUri($name, $parameters = [], $relative = false)
...
+
         return sprintf(
-            '%s%s?v=%s',
+            '%s%s%s',
```
Comment (Maks3w) [inline src/TwigExtension.php]: Probably its better throw away `sprintf` and use the concat operator

File: src/TwigExtension.php (hunk 1 above)
Comment (geerteltink) [inline src/TwigExtension.php]: I tried but it didn't work. From the 2 tests I get this:

testRenderAssetUrlUsesComposedAssetUrlAndVersionToGenerateUrl:
0 = XYZ | NULL

Should be XYZ

testRenderAssetUrlUsesProvidedVersionToGenerateUrl
Y[[ = XYZ | ABC

Should be  ABC

**Comment by geerteltink** [comment]
@Maks3w Thanx for the comments. Fixed where possible.

File: src/TwigExtension.php (hunk 1 above)
Comment (Maks3w) [inline src/TwigExtension.php]: I was confused that trick is for JavaScript.

Comment (Maks3w) [inline src/TwigExtension.php]: The equivalent for PHP is `$version ?: $this->assetsVersion`

**Comment by Maks3w** [comment]
Should be `0` allowed as a valid version?

**Comment by Maks3w** [comment]
The assetsVersion parameter in the constructor method should be optional.

**Comment by geerteltink** [comment]
> Should be 0 allowed as a valid version?

Doesn't matter what you want as a version. That's up to the developer. If you want it `0` or `XYZ`. Doesn't matter. I guess the only restriction could be that it is a valid url parameter. Maybe we could url_encode it, but then again, the developer should be warned when something is wrong.

> The assetsVersion parameter in the constructor method should be optional.

Actually that doesn't matter. It is always passed from the TwigRendererFactory. It's either the value from the config or an empty string:
https://github.com/zendframework/zend-expressive-twigrenderer/blob/master/src/TwigRendererFactory.php#L64-L68

**Comment by Maks3w** [comment]
I say `0` because with the current code and proposed changes `"0"` will be evaluated as false

**Comment by geerteltink** [comment]
Ah, you are right. Well, if someone wants to have a `0` version I guess it could be considered the same as empty and thus not displaying the version string.

**Pull request merged**